xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp

```diff
@@ -367,7 +367,7 @@ std::atomic<bool> CDVDVideoCodecAndroidMediaCodec::m_InstanceGuard(false);
 
 bool CDVDVideoCodecAndroidMediaCodec::Open(CDVDStreamInfo &hints, CDVDCodecOptions &options)
 {
-  int num_codecs;
+  std::vector<CJNIMediaCodecInfo> codecInfos;
```

Comment:
This is C++. Please declare and init the variable, where it first is needed.

Reply:
The rest of the variable declarations I've added are where they are initialised and make sense.

This case is special because the code has several ```GOTO statements```. If the definition is where it should be, the error "jump bypasses variable initialization" is generated.

For the moment and to make the minimum changes, I decided to declare the variable where one of the variables it replaces was already defined, although I didn't like it either.

Reply:
ugh, `goto FAIL;` ?!?
can we (maybe in a follow up) define a new private member `bool DoCleanupAndFail() {...}` and just call `return DoCleanupAndFail();` instead?

Reply:
Better to manage it in a separate PR, do not mix these two issues.